[REF] web_editor: use the Notebook component

Iucapad opened the pull request:
This commit makes use of the Notebook component in the MediaDialog component. Since odoo#98706, the design used by tabs and the Notebook is more consistent, which allow us to replace that part of the template by the component.

The Notebook receives the list of pages to render, and will generate the right subcomponents in the content part of its template.

=> Less code, same functionality and behavior

robodoo commented:
[Pull request status dashboard](https://mergebot.odoo.com/odoo/odoo/pull/100253).

Iucapad commented:
Hey @younn-o, here is a PR for the change I suggested last week. Let me know what you think.
It should still work as expected, since the default page is correctly given as a props, and the Notebook DOM is the same after all

Would you review this, or should I ask to someone else? Thank you

younn-o commented on addons/web_editor/static/src/components/media_dialog/media_dialog.xml:
```diff
@@ -4,21 +4,7 @@
-                </t>
-            </div>
-        </div>
+        <Notebook pages="tabs" onPageUpdate.bind="onTabChange" defaultPage="state.activeTab"/>
```
Funny enough, it solves an issue I was fixing differently here https://github.com/odoo/odoo/pull/100039/commits/6b977f90f2025fd28e5aced9596f3164810ca232 (in the end, it's the same fix, because only the active page is rendered).
Other than that there are a few changes in the style (`bg-white pt-3` is added on the notebook content, so the icons are black on the icons tab, and on mobile it does not behave the same way), maybe @qsm-odoo can have a look at it ?

younn-o replied:
I tested again and my bad, it's the same behaviour in mobile, sorry for that: I tested with the dev tools, resizing to get to mobile dimensions but that never happens. On mobile, starting the media dialog is the same with the Notebook component or without !

qsm-odoo reviewed (changes requested):
Commit message:
- As said in my other comment: adapt to use part of Younn's commit message too
- FIX instead of REF
- 72 characters (max) long lines for the commit message (not ~55)
- Instead of `Since odoo#98706` mention the commit that was merged in that PR: easier to get to the actual diff that was merged.
- I would also suggest always using full github URL when mentioning github resources so that users using `git log` in their own console can actually click on the link.

Thanks for the fix :+1:

younn-o commented on addons/web_editor/static/src/components/media_dialog/media_dialog.js:
```diff
@@ -127,6 +128,10 @@ export class MediaDialog extends Component {
         }
     }
 
+    onTabChange(tab) {
+        this.state.activeTab = tab;
+    }
```
I think for now the website components methods are not following specific guidelines, besides maybe having the getters and setters first, but they could be organized better for sure

qsm-odoo replied:
Ok, let's put this one at end despite of the rest of the organization @Iucapad if you don't mind, thanks :+1:

Iucapad commented:
Hello, I applied some changes as you suggested. As for the BS classes on the notebook content, I'm not sure why they were added in this commit: https://github.com/odoo/odoo/commit/68f1c90a031a7c2f751d9c8eca7ff525e9f7e206.

Cc @anso-odoo and @stefanorigano :
Is the bg-white related to the upcoming dark mode, or could we remove it?
And is the pt-3 really needed on the notebook content?

stefanorigano commented:
Hello @Iucapad ,

> Cc @anso-odoo and @stefanorigano : Is the bg-white related to the upcoming dark mode, or could we remove it?

Confirmed, we can removed it (actually we removed it in the dark-mode branch too)

> And is the pt-3 really needed on the notebook content?

Can be removed as well. There is a minor inconsistency between frontent/backend that handle spacing on `tab-pane` using margin and padding respectively... nothing critical though, we can refine it later.

Iucapad commented:
Okay thanks!

So, @qsm-odoo, what do you think of the current state of the PR?
I changed the commit following your advices and added another commit to remove the classes from the component

qsm-odoo reviewed (approved):
Thanks @Iucapad, seems good :+1: 

@robodoo rebase-ff r+
